src/utils/storageFactory.ts

```diff
@@ -0,0 +1,41 @@
+import { checkIsJson } from './checkIsJson';
+
+const storageFactory = (storage: Storage) => {
+  const getItem = <D>(key: string, defaultValue: D | unknown) => {
+    try {
+      const storedValue = storage.getItem(key);
+      if (storedValue) {
+        return checkIsJson(storedValue) ? JSON.parse(storedValue) : storedValue;
+      }
+      return defaultValue;
+    } catch (e) {
+      if (e instanceof Error) {
+        alert(`${storage}에서 아이템 꺼내는 도중 오류 발생 ${e.message}`);
+      }
+    }
+  };
+
+  const setItem = <V>(key: string, value: V | unknown) => {
+    try {
+      storage.setItem(key, JSON.stringify(value));
+    } catch (e) {
+      if (e instanceof Error) {
+        alert(`${storage}에 아이템 넣는중 도중 오류 발생 ${e.message}`);
+      }
+    }
+  };
+
+  const removeItem = (key: string) => {
+    try {
+      storage.removeItem(key);
+    } catch (e) {
+      if (e instanceof Error) {
+        alert(`${storage}에서 아이템 삭제하는중 오류 발생 ${e.message}`);
+      }
+    }
```

Comment:
removeItem의 경우 [멱등성](https://stackoverflow.com/questions/31448160/check-if-localstorage-removeitem-was-successful)을 가지고 있어서 try-catch로 안감싸도 괜찮을 것 같습니다! 
```suggestion
      storage.removeItem(key);
    }
```